internal/pkg/api/handleAck.go

```diff
@@ -385,8 +377,24 @@ func (ack *AckT) handlePolicyChange(ctx context.Context, zlog zerolog.Logger, ag
 	return errors.Wrap(err, "handlePolicyChange update")
 }
 
+func (ack *AckT) invalidateAPIKeys(ctx context.Context, agent *model.Agent) {
+	var ids []string
+	for _, out := range agent.Outputs {
+		for _, k := range out.ToRetireAPIKeyIds {
```

Comment:
Q: when this is cleared?

Reply:
below, within the [`if`](https://github.com/elastic/fleet-server/pull/1684/files#diff-5d9f1022109ff4fde11455f4c1bb063797e4fde8a3ad0ff732f560bc5ab08f83R389)